lib/kms/common.js

```diff
@@ -0,0 +1,122 @@
+import crypto from 'crypto';
+import { errors } from 'arsenal';
+
+export default class common {
+
+    static _algorithm() {
+        return 'aes-256-ctr';
+    }
+
+    /* AES-256 Key */
+    static _keySize() {
+        return 32;
+    }
+
+    /* IV is 128bit for AES-256-CTR */
+    static _IVSize() {
+        return 16;
+    }
+
+    /* block size is 128bit for AES-256-CTR */
+    static _aesBlockSize() {
+        return 16;
+    }
+
+    static createDataKey() {
```

Comment:
Let's add a comment:
Creates data key to encrypt and decrypt the actual data (which data key is ciphered and deciphered by the appliance).  The encrypted data key is stored in the object's metadata. We also use this function to create the bucket key for the file and in memory implementations.